yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
                 "'{}' is unrecognized ranking method".format(algorithm)
             )
 
+        # Extract matrix from dataframe if necessary
+        if is_dataframe(X):
```

Comment:
If someone specifies a list of features to rank (e.g. a subset of the total features) in both the dataframe or the named object array, it looks like this is the point where the filtering has to be done. Not sure if you did that further down, but it's something to consider.